Mathlib/FieldTheory/Perfect.lean

```diff
@@ -231,22 +230,11 @@ instance toPerfectRing (p : ℕ) [ExpChar K p] : PerfectRing K p := by
   rw [g.natDegree_map ι, ← Nat.pos_iff_ne_zero, natDegree_pos_iff_degree_pos]
   exact minpoly.degree_pos ha
 
-theorem separable_iff_squarefree {g : K[X]} : g.Separable ↔ Squarefree g := by
-  refine ⟨Separable.squarefree, ?_⟩
-  induction' g using UniqueFactorizationMonoid.induction_on_coprime with p hp p n hp p q _ ihp ihq
-  · simp
-  · obtain ⟨x, hx, rfl⟩ := isUnit_iff.mp hp
-    exact fun _ ↦ (separable_C x).mpr hx
-  · intro hpn
-    rcases hpn.eq_zero_or_one_of_pow_of_not_isUnit hp.not_unit with rfl | rfl; · simp
-    exact (pow_one p).symm ▸ PerfectField.separable_of_irreducible hp.irreducible
-  · intro hpq'
-    obtain ⟨h, hp, hq⟩ := squarefree_mul_iff.mp hpq'
-    apply (ihp hp).mul (ihq hq)
-    classical
-    apply EuclideanDomain.isCoprime_of_dvd
-    · rintro ⟨rfl, rfl⟩; simp at hp
-    · exact fun d hd _ hdp hdq ↦ mem_nonunits_iff.mpr hd <| h d hdp hdq
+theorem separable_iff_squarefree {g : K[X]} : g.Separable ↔ Squarefree g :=
+  ⟨Separable.squarefree, fun sqf ↦ isCoprime_of_irreducible_dvd (sqf.ne_zero ·.1) fun p h ↦ by
+    rintro ⟨q, rfl⟩ dvd
+    rw [derivative_mul, dvd_add_left (dvd_mul_right p _)] at dvd
+    exact h.1 (sqf _ <| mul_dvd_mul_left _ <| (separable_of_irreducible h).dvd_of_dvd_mul_left dvd)⟩
```

Comment:
This is a great proof but I think we should make it a bit to read:
```suggestion
theorem separable_iff_squarefree {g : K[X]} : g.Separable ↔ Squarefree g := by
  refine ⟨Separable.squarefree,
    fun sqf ↦ isCoprime_of_irreducible_dvd (not_and_of_not_left _ sqf.ne_zero) ?_⟩
  rintro p (h : Irreducible p) ⟨q, rfl⟩ (dvd : p ∣ derivative (p * q))
  suffices IsUnit p by exact h.1 this
  replace dvd : p ∣ q := by
    rw [derivative_mul, dvd_add_left (dvd_mul_right p _)] at dvd
    exact (separable_of_irreducible h).dvd_of_dvd_mul_left dvd
  exact sqf _ <| mul_dvd_mul_left _ dvd
```

Reply:
Thanks! I mostly adopted your suggestion in [c79c948](https://github.com/leanprover-community/mathlib4/pull/10236/commits/c79c948495b7c59bc57ac4f5f7711a7c5c945e9f) except I combined the `suffices` line with the last `exact` as a compromise between verbosity and readability. (I also think `(sqf.ne_zero ·.1)` is nice; these side condition should ideally be automated away rather than showing up in the proof, so it's desirable to make its appearance as short as possible.)

Reply:
I restored `(sqf.ne_zero ·.1)` since you thumbed up.